examples/ssd_mobilenet/utils.js

```diff
@@ -0,0 +1,504 @@
+    // NHWC layout
+    for (let y = 0; y < height; ++y) {
+      for (let x = 0; x < width; ++x) {
+        for (let c = 0; c < channels; ++c) {
+          let value = pixels[y*width*imageChannels + x*imageChannels + c];
+          tensor[y*width*channels + x*channels + c] = (value - mean)/std;
+        }
+      }
+    }
+  }
+
+  /**
+  * Decode out box coordinate
+  *
+  * @param {Float32Array[4]} offsets - An 4 element Float32Array of box shape offsets.
+  * @param {number[4]} anchorCord - An 4 element Array of anchor coordinate.
+  * @returns {number[4]} decodeBox - An 4 element Float32Array of box decode shape.
+  */
+  _box_decode(offsets, anchorCord) {
+    if (offsets.length !== 4 || anchorCord.length !== 4) {
+      throw new Error('[box_decode] Each input length should be 4!');
+    }
+    const [ycenter_a, xcenter_a, ha, wa] = anchorCord;
+    let [ty, tx, th, tw] = offsets;
+
+    // scale_factors = [y_scale, x_scale, height_scale, width_scale]
+    let scale_factors = [10.0, 10.0, 5.0, 5.0];
+    ty /= scale_factors[0];
+    tx /= scale_factors[1];
+    th /= scale_factors[2];
+    tw /= scale_factors[3];
+
+    let w = Math.exp(tw) * wa;
+    let h = Math.exp(th) * ha;
+    let ycenter = ty * ha + ycenter_a;
+    let xcenter = tx * wa + xcenter_a;
+    let ymin = ycenter - h / 2;
+    let xmin = xcenter - w / 2;
+    let ymax = ycenter + h / 2;
+    let xmax = xcenter + w / 2;
+    let decodeBox = [ymin, xmin, ymax, xmax];
+    return decodeBox;
+  }
+
+  /**
+  * Decode output tensor boxes
+  * 
+  */
+  _outputTensorDecode() {
+    let boxOffset = 0;
+    for (let y = 0; y < NUM_BOXES; ++y) {
+      let outCord = this.outputBoxTensor.subarray(boxOffset, boxOffset + BOX_SIZE);
+      outCord.set(this._box_decode(outCord, this.anchors[y]));
+      boxOffset += BOX_SIZE;
+    }
+  }
+
+  /**
+  * Get IOU(intersection-over-union) of 2 boxes
+  *
+  * @param {number[4]} boxCord1 - An 4 element Array of box coordinate.
+  * @param {number[4]} boxCord2 - An 4 element Array of box coordinate.
+  * @returns {number} IOU
+  */
+  _IOU(boxCord1, boxCord2) {
+    if (boxCord1.length !== 4 || boxCord2.length !== 4) {
+      throw new Error('[box_decode] Each input length should be 4!');
+    }
+
+    const [ymin1, xmin1, ymax1, xmax1] = boxCord1;
+    const [ymin2, xmin2, ymax2, xmax2] = boxCord2;
+    let minYmax = Math.min(ymax1, ymax2);
+    let maxYmin = Math.max(ymin1, ymin2);
+    let height = Math.max(0, minYmax - maxYmin);
+    let minXmax = Math.min(xmax1, xmax2);
+    let maxXmin = Math.max(xmin1, xmin2);
+    let width = Math.max(0, minXmax - maxXmin);
+    let intersection = height * width;
+    let area1 = (ymax1 - ymin1) * (xmax1 - xmin1);
+    let area2 = (ymax2 - ymin2) * (xmax2 - xmin2);
+    let areaSum = area1 + area2 - intersection
+    if (areaSum === 0) {
+      throw new Error('[IOU] areaSum can not be 0!');
+    }
+    let IOU = intersection / areaSum;
+    return IOU;
+  }
+
+  /**
+  * Generate anchors
```

Comment:
Please add comment by referring to `create_ssd_anchors` of https://github.com/tensorflow/models/blob/master/research/object_detection/anchor_generators/multiple_grid_anchor_generator.py

Reply:
And please add referring to `ssd_anchor_generator` of https://github.com/tensorflow/models/blob/master/research/object_detection/samples/configs/ssd_mobilenet_v1_coco.config